### ISSUE TITLE
Audio block not handled correctly when surrounded by <strong> tags

### ISSUE THREAD

**Issue by mattford** [xemlock/htmlpurifier-html5#17]
With the below code, the <source> tag is stripped from the audio block and replaced with <strong></strong>

```
<?php
require_once('vendor/autoload.php');

$html = '<p><strong><audio controls="controls"><source type="audio/mp3" src="myaudiofile.mp3" /></audio></strong></p>';

echo "In: " . $html . PHP_EOL;

$config = HTMLPurifier_HTML5Config::createDefault();
$purifier = new HTMLPurifier($config);


echo "out: " . $purifier->purify($html);
```

Expected output:

`<p><strong><audio controls><source type="audio/mp3" src="myaudiofile.mp3" /></audio></strong></p>`

(Unless I'm reading it wrong, the spec says `<strong>` can contain "Phrasing content" which includes `<audio>`) http://w3c.github.io/html/single-page.html#phrasing-content-2

Actual output:

`<p><strong></strong></p><audio controls><strong></strong></audio><strong></strong>`

I think it's because the `<strong>` tag in the base library is set to allow contents of type "Inline", whereas `<audio>` is defined as a block in this library.

Will follow up with a PR if I find a fix today

**Comment by xemlock** [comment]
@mattford You're right - the content of `$html` variable is a valid HTML5 (confirmed with https://html5.validator.nu/). Thanks for reporting.

**Comment by xemlock** [comment]
@mattford FYI #18. Feel free to comment/review.

**Comment by mattford** [comment]
Looks like the following elements in this lib need to be Inline:

`video, audio, picture, del (already Inline in base lib)`

as line 47 of `HTMLPurifier\HTMLModule\Text.php` of the base library defines `<strong>` as containing inline elements only:

`$strong = $this->addElement('strong', 'Inline', 'Inline', 'Common');`

I did notice however that there are some elements defined as inline in the base library which are invalid within a `<strong>` tag (but this is a bug in HTMLPurifier rather than your HTML5 extension I believe)

`baseFont, font, strike, big, i, small, tt, marquee, acronym`

There is one element defined in your library which is not valid inside `<strong>`:

`figcaption`

I suspect this might be another bug as the only permitted parent of this element is `figure`, which has its contents defined as `Flow`

**Comment by mattford** [comment]
@xemlock PR looks good

**Comment by xemlock** [comment]
Thanks for review! I will fix content category for `<picture>` and `<del>` as well. 

As for `<figcaption>`, according to [spec](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption), it can only appear as a direct child of `<figure>`. So I see no error here.

**Comment by mattford** [comment]
What I was confused about with `figcaption` is that the `figure` is defined to have contents of `Flow`, whereas the type of `figcaption` is `Inline`, but I tested it and it works. 

No doubt you have more in depth knowledge of this stuff than I do, so I probably missed something there!

**Comment by xemlock** [comment]
@mattford Here is a quick explanation how it works:

```php
$def->addElement('figure', 'Block', new HTMLPurifier_ChildDef_Figure(), 'Common');
$def->addElement('figcaption', false, 'Flow', 'Common');
```

`false` means that `<figcaption>` cannot appear directly in the body - its appearance is controlled by the `HTMLPurifier_ChildDef_Figure` content model.

`'Flow'` is the content category of child elements that `<figcaption>` can contain.

`'Common'` means that `<figcaption>` inherits common attributes.

**Comment by mattford** [comment]
Thanks for the explanation, makes sense now.

**Comment by xemlock** [comment]
Fixed by #18 and #19.
